### PR TITLE
Update `.filter()` docs

### PR DESCRIPTION
**Pull request by markfairbanks**
Wanted to try and start filling in some documentation. I started with one function for now just to make sure this is all formatted correctly.

@ritchie46 - do you want the same examples added for `pl.LazyFrame.filter`?

### CODE REVIEW

File: py-polars/polars/eager/frame.py [hunk 1]
```diff
@@ -1252,6 +1252,39 @@ def filter(self, predicate: "pl.Expr") -> "DataFrame":
...
+        Examples
+        --------
+
+        >>> df = pl.DataFrame({
+        >>>     "foo": [1, 2, 3],
+        >>>     "bar": [6, 7, 8],
+        >>>     "ham": ['a', 'b', 'c']
+        >>> })
+        >>> # Filter on one condition
+        >>> df.filter(col("foo") < 3)
```
Comment (ritchie46) [inline py-polars/polars/eager/frame.py:1265]: Maybe use `pl.col` for copy/pase ability.

**Review by ritchie46 — APPROVED** [review]

**Comment by ritchie46** [comment]
Yes, it looks good. Thanks!

> @ritchie46 - do you want the same examples added for pl.LazyFrame.filter?

That would be great. The lazy documentation could use a boost

File: py-polars/polars/lazy/frame.py [hunk 2]
```diff
@@ -388,6 +388,39 @@ def filter(self, predicate: "Expr") -> "LazyFrame":
...
+        Examples
+        --------
+
+        >>> df = pl.DataFrame({
+        >>>     "foo": [1, 2, 3],
+        >>>     "bar": [6, 7, 8],
+        >>>     "ham": ['a', 'b', 'c']
+        >>> }).lazy()
+        >>> # Filter on one condition
+        >>> df.filter(pl.col("foo") < 3).collect()
```
Comment (ritchie46) [inline py-polars/polars/lazy/frame.py:401]: Shall we name the variable differently? I always think of a `DataFrame` when reading `df`.

`LazyFrame -> lf` ?

Comment (markfairbanks) [reply py-polars/polars/lazy/frame.py:401]: Yep, makes sense. I'll make the change

**Comment by ritchie46** [comment]
Thanks @markfairbanks!

**Pull request merged**